moesif_aws_lambda/middleware.py

```diff
@@ -391,17 +400,34 @@ def after(self, retval):
                 if self.DEBUG:
                     print('Moesif Event Model:')
                     print(json.dumps(self.event))
-                
-                if self.DEBUG:
-                    start_time_sending_event_w_rsp = datetime.utcnow()
-                    event_send = self.api_client.create_event(event_model)
-                    end_time_sending_event_w_rsp = datetime.utcnow()
+
+                # Sampling Rate
+                random_percentage = random.random() * 100
+
+                self.sampling_percentage = self.app_config.get_sampling_percentage(event_model,
+                                                                                   self.config,
+                                                                                   self.user_id,
+                                                                                   self.company_id)
+
+                if self.sampling_percentage >= random_percentage:
+                    event_model.weight = 1 if self.sampling_percentage == 0 else math.floor(
+                        100 / self.sampling_percentage)
+
                     if self.DEBUG:
-                        print("[moesif] Time took in sending event to moesif in millisecond - " + str(get_time_took_in_ms(start_time_sending_event_w_rsp, end_time_sending_event_w_rsp)))
-                        print('[moesif] Event Sent successfully ' + str(event_send))
+                        start_time_sending_event_w_rsp = datetime.utcnow()
+                        event_send = self.api_client.create_event(event_model)
+                        end_time_sending_event_w_rsp = datetime.utcnow()
+                        if self.DEBUG:
+                            print("[moesif] Time took in sending event to moesif in millisecond - " + str(
+                                get_time_took_in_ms(start_time_sending_event_w_rsp, end_time_sending_event_w_rsp)))
+                            print('[moesif] Event Sent successfully ' + str(event_send))
+                    else:
```

Comment:
this does not look correct. Why it should create event if debug is false?

Reply:
> this does not look correct. Why it should create event if debug is false?

We created the event both when debug is `true`(line 418) and `false`(line 425). The different is we retain the `event_send` and log it when debug is `true`

Reply:
fixed the blocking/unblocking call based on refresh update time